src/builtins/utils.zig

```diff
@@ -12,22 +12,95 @@ const DEBUG_INCDEC = false;
 const DEBUG_TESTING_ALLOC = false;
 const DEBUG_ALLOC = false;
 
+/// TODO
+pub const RocAlloc = *const fn (size: usize, alignment: u32) callconv(.C) ?*anyopaque;
+/// TODO
+pub const RocDealloc = *const fn (c_ptr: *anyopaque, alignment: u32) callconv(.C) void;
+
+/// Tracks allocations for testing purposes with C ABI compatibility. Uses a single global testing allocator to track allocations. If we need multiple independent allocators we will need to modify this and use comptime.
+pub const TestAllocator = struct {
```

Comment:
If you look at this test allocator for example, the entire struct could be the `context: anyopaque` as an example. That said, the feature is specifically need for arenas and local allocators.